### ISSUE TITLE
How to use train and test split with the recipes? 

### ISSUE THREAD

**Issue by 7rabbit** [pytorch/torchtune#2222]
Dear torchtune team,

With sft trainer we can do 
train_dataset=ds["train"],
eval_dataset=ds["validation"],
if it is a split from huggingface dataset.

I wonder under a fine-tuning recipes with instruction dataset, how is this achieved, particular in a YAML configuration file? With current example on the tutorial: split: train, I feel that the whole dataset is used for training. Should we prepare json/csv files before hand with spitted on train/test/validation set?
Thanks

**Comment by joecummings** [comment]
Hey @7rabbit - currently this isn't available through a YAML config. You're more than welcome to hack onto our recipes to add this functionality, but we also have it on our roadmap to support early this year!

For now, if you want to only train on part of the dataset you can either preprocess yourself, do it "online" through a custom transform, or specify a percentage of the dataset to use like so "train[:50%]".

**Comment by shaunakjoshi12** [comment]
Hi @joecummings asking for a follow up if this feature has been implemented for the yaml config?

**Comment by ebsmothers** [comment]
@shaunakjoshi12 we have not yet implemented this feature. If you're interested in contributing it we would be happy to review a PR!

**Comment by habibregask86** [comment]
any news on this feature please ?
